### PR TITLE
Postmaster support

### PR DESCRIPTION
**Pull request by acalatrava**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -38,6 +38,7 @@ require (
 	go.etcd.io/bbolt v1.3.5
 	golang.org/x/crypto v0.0.0-20200728195943-123391ffb6de
 	golang.org/x/sync v0.0.0-20201020160332-67f06af15bc9
+	golang.org/x/text v0.3.2
```
Comment (jaytaph) [inline go.mod:41]: is this import needed?

Comment (acalatrava) [reply go.mod:41]: actually I don't know where is that coming from.... :-/

File: cmd/bm-server/processor/retry_queue.go [hunk 2]
```diff
@@ -42,10 +48,46 @@ func ProcessRetryQueue(forceRetry bool) {
 	for _, info := range retryQueue {
 		if info.Retries > MaxRetries {
 			// @TODO: We should send a message back to the user?
+			logrus.Trace("Send a postmaster message")
```
Comment (jaytaph) [inline cmd/bm-server/processor/retry_queue.go:51]: could we move the "send message" functionality to a separate function? This keeps the complexity lower and we might be able to use it for other purposes later too

File: internal/message/catalog.go [hunk 3]
```diff
@@ -124,6 +125,23 @@ func NewCatalog(sender, recipient *address.Address, subject string) *Catalog {
...
+// NewServerCatalog initialises a new (server) catalog. This catalog has to be filled with more info, blocks and attachments
+func NewServerCatalog(sender, recipient *hash.Hash, subject string) *Catalog {
+	c := &Catalog{}
+
+	c.CreatedAt = time.Now()
+	c.From.Address = sender.String()
+	c.From.Name = "Postmaster"
+	c.From.Address = sender.String()
+	c.To.Address = recipient.String()
+	c.AddFlags("postmaster")
```
Comment (jaytaph) [inline internal/message/catalog.go:137]: first actual use of the flags. Noice!

File: cmd/bm-server/processor/retry_queue.go (hunk 2 above)
Comment (acalatrava) [reply cmd/bm-server/processor/retry_queue.go:51]: absolutely

File: internal/message/upload.go [hunk 4]
```diff
@@ -262,3 +263,40 @@ func MoveMessage(srcSection Section, targetSection Section, msgID string) error
 
 	return fs.Rename(p1, p2)
 }
+
+// StoreLocalMessage will store a message locally
+func StoreLocalMessage(header *Header, catalog io.Reader, blocks map[string]*io.Reader, attachments map[string]*io.Reader) (string, error) {
```
Comment (jaytaph) [inline internal/message/upload.go:268]: Could we use an envelope here instead of sending all these arguments (not sure if that's possible though)

Comment (acalatrava) [reply internal/message/upload.go:268]: definitely! ouch!

**Review by jaytaph — APPROVED** [review]
lgtm

**Pull request merged**